### PR TITLE
Add additional loging to max_pods e2e test

### PR DESCRIPTION
**Pull request by gmarek**
Add some logging to help with debugging #10720

### CODE REVIEW

**Comment by davidopp** [comment]
It's also fine to just merge it as-is, I think. We can fix the minor comments I had another time.

**Comment by davidopp** [comment]
Sorry, previous comment went to wrong PR. I will review this PR in a second.

File: test/e2e/max_pods.go [hunk 1]
```diff
@@ -100,6 +100,12 @@ var _ = Describe("MaxPods", func() {
 		By(fmt.Sprintf("Starting additional %v Pods to fully saturate the cluster and trying to start a new one", podsNeededForSaturation))
 		expectNoError(RunRC(config))
 
+		// Make sure that cluster is saturated
```
Comment (davidopp) [inline test/e2e/max_pods.go]: The comment says "make sure" but really you just log the number. Would it be better to compare len(pods.Items) and totalPodCapacity and fail the test if they're != ? Or at least only print a log message if they are != ?

Comment (gmarek) [inline test/e2e/max_pods.go]: This is what's happening right now (kind of). What I want to do is to check if failures are really caused by some pods being unhealthy in some way.

Comment (davidopp) [inline test/e2e/max_pods.go]: IIUC, you're currently checking that after you try to schedule the max+1 pod, there are max pods running. Which isn't quite the same as checking that before you try to schedule the max+1 pod, there are max pods running.

Anyway, it's not a big deal.

**Comment by davidopp** [comment]
LGTM

**Pull request merged**